### DISCUSSION TITLE
Support for round-trip import/export of YAMLs

### DISCUSSION THREAD

**Discussion by monsdar** [tobiasge/netbox-initializers#40] [Ideas]
I like the idea of having YAMLs in a Git repo to init netbox with. To even go further down the road of IaC I'd like to ask if there's a way to export YAMLs as well. That would allow for roundtrip-engineering:
1. Initially provide a repo with few YAMLs to set up Regions, Sites, Locations
1. Have users enter more data regarding Racks, Devices etc
1. Export data from Netbox as YAMLs and push them to the repo

Which allows the following usecases:
1. Users can now enter data directly in Netbox or extend the YAMLs in the repo
1. Each new deployment of netbox (staging area, new cluster, ...) is easy to init with the latest data
1. New Deployments (sister companies, other departments, ...) can take from the existing data what's useful and init their own netbox with it

I started this as a conversation because I'm not aware if there a) already is a solution similar to this and if b) there is a common interest of driving this approach further.

**Comment by tobiasge** [comment]
Netbox has its own export and import methods. For this plugin export is out of scope.
